src/util/bidi.js

```diff
@@ -198,15 +198,16 @@ let bidiOrdering = (function() {
       order.push(new BidiSpan(0, len - m[0].length, len))
     }
 
-    return order
+    order.forDirection = direction
+    return direction == "rtl" ? order.reverse() : order
```

Comment:
I wouldn't put 'wild' properties on the array -- since global direction changes are probably rare, you could clean out the `order` properties on all lines when they happen.